principles.rst

```diff
@@ -303,3 +309,124 @@ down GHC even when those features are not active. Yet this principle is importan
+* Depends only on stable Haskell packages.
+
+Boot packages shipped with GHC will be designated as stable or not in the documentation:
+
+* ``base`` is stable, despite depending on unstable packages.
+* Any GHC-internal packages (e.g. ``ghc-internal``, ``ghc-prim``, ``ghc-bignum``) are not stable and may change API without notice and between minor releases.
+* The ``ghc`` package is not stable. (There is a separate project to define a more stable GHC API, but that is out of scope for this document.)
+* ``ghc-experimental`` is not stable, but should aim to minimise breaking changes as far as feasible, since it is user-facing.
+* ``template-haskell`` is not stable, because it contains AST datatypes which need to be changed as the language evolves.
+
+Stability (GR1)
+~~~~~~~~~~~~~~~~
+
+**General rule (GR1)**.  *A stable Haskell package that works today should continue to work in subsequent releases of GHC (using the same compiler flags of course).*
+
+Notes and clarifications:
+
+* The general goal of "works" includes both "compiles successfully" and "runs successfully, as fast as it did before".  That is a very demanding goal, so it has the status of a highly-sought-after aspiration rather than an unbreakable promise.
+
+* *The base library*.  Achieving (GR1) depends crucially on the stability of the base package; but that is already under the careful management of the Core Libraries Committee.  However ``base`` does, and should, evolve, so achieving (GR1) really depends on achieving the "reinstallable ``base``" goal.  That is, you should be able to compile a package with (say) GHC 9.16, using a version of ``base`` that exposes precisely the API of the ``base`` that came with GHC 9.12.
+
+* *Warnings*
+
+  * ``-Werror`` is excluded because otherwise (GR1) would be broken whenever we add a warning.
+  * Similarly ``-Werror=flag`` is excluded to allow warnings to fire in more cases without breaking (GR1).
+  * It's fine for a stable package to use ``-Werror`` in a CI build, to help the author find warnings.  But not in the default configuration, used by others when installing the package.
+  * There is no stability guarantee that a later GHC will emit the same warnings as an earlier GHC.  A notable case in point is deprecations, where a later GHC may advise authors to (say) import a function from a different module; while an earlier GHC obviously will not.  In general, warnings should not be regarded as stable.
+
+* *Language editions*.
+
+  * A stable package should specify an explicit language edition because subsequent releases of GHC might change the default language edition.  So if the package does not pin a specific language edition, it might then fail when compiled with a later release.
+  * Language editions like ``GHC202x`` should use only Stable extensions.
+
+* (GR1) is restricted to Stable extensions, and excludes dependence on ``ghc-experimental``.  That leaves the "experimental" space open to more rapid change.
```

Comment:
I think we can avoid the need to define a "Stable" category, and simply talk about experimental or not. That avoids making assumptions about how #601 will be resolved.

```suggestion
  * Language editions like ``GHC202x`` should not use experimental extensions.

* (GR1) is restricted to non-experimental extensions, and excludes dependence on ``ghc-experimental``.  That leaves the "experimental" space open to more rapid change.
```